vllm/v1/structured_output/backend_xgrammar.py

```diff
@@ -94,15 +96,26 @@ def __init__(self, vllm_config: VllmConfig):
         )
 
     def _get_grammar_compiler_threads(self) -> int:
-        """Get the number of threads to use for the grammar compiler. Half
-        of the total thread count is chosen here to avoid occupying all CPU
-        resources. 32 is an estimate of the effective number of compilation
-        threads needed for common schemas, and using more threads may not
-        provide additional benefits.
+        """Get the number of threads to use for the grammar compiler. If the
+        environment variable VLLM_XGRAMMAR_COMPILER_THREADS is set, it will
+        be used as the number of threads. Otherwise, half of the total thread
+        count is chosen here to avoid occupying all CPU resources. 32 is an
+        estimate of the effective number of compilation threads needed for
+        common schemas, and using more threads may not provide additional
+        benefits.
 
         Returns:
             int: The number of threads to use for the grammar compiler.
         """
+        # check environment variable
+        env_threads = os.environ.get(VLLM_XGRAMMAR_COMPILER_THREADS_ENV_VAR)
```

Comment:
We have another pattern we typically use for env vars in vLLM. I'm happy to do the conversion for you if you don't mind me committing to your branch. See `vllm.envs`